src/Mod/Arch/importOBJ.py

```diff
@@ -72,14 +72,23 @@ def getIndices(obj,shape,offsetv,offsetvn):
             try:
                 if not isinstance(e.Curve,Part.LineSegment):
                     if not curves:
-                        myshape = obj.Shape.copy(False)
-                        myshape.Placement=obj.getGlobalPlacement()
+                        if obj.isDerivedFrom("App::Link"):
+                           myshape = obj.LinkedObject.Shape.copy(False)
+                           myshape.Placement=obj.LinkPlacement
+                        else:
+                           myshape = obj.Shape.copy(False)
+                           myshape.Placement=obj.getGlobalPlacement()
                         mesh=MeshPart.meshFromShape(Shape=myshape, LinearDeflection=0.1, AngularDeflection=0.7, Relative=True)
                         FreeCAD.Console.PrintWarning(translate("Arch","Found a shape containing curves, triangulating")+"\n")
                         break
             except: # unimplemented curve type
-                myshape = obj.Shape.copy(False)
-                myshape.Placement=obj.getGlobalPlacement()
+                if obj.isDerivedFrom("App::Link"):
+                  if obj.Shape:
+                   myshape = obj.Shape.copy(False)
+                   myshape.Placement=obj.LinkPlacement
+                  else:
+                   myshape = obj.Shape.copy(False)
+                   myshape.Placement=obj.getGlobalPlacement()
```

Comment:
Please add proper 4 space indentation in the code that you are modifying. Also, please be aware that now the Draft and Arch workbenches have coding standards. Basically, "follow PEP8".

Reply:
Sorry, I am definitely not a python developper (C++ or Golang) and do my best.

Reply:
Awesome. We need more C++ programmers because that's where the real meat is, Part and PartDesign.